### PR TITLE
Drop check for "Old-style" components

### PR DESCRIPTION
**Pull request by corvus-ch**
## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update tests.
- [x] Update the ./CHANGELOG.md.

### CODE REVIEW

File: commodore/dependency_mgmt.py [hunk 1]
```diff
@@ -25,20 +25,11 @@ def create_component_symlinks(cfg, component: Component):
...
-            P("inventory/classes/defaults"),
-            dest_name=f"{component.name}.yml",
-        )
-    else:
-        click.secho(
-            "     > Old-style component detected. Please move "
-            + "component defaults to 'class/defaults.yml'",
-            fg="yellow",
-        )
+    relsymlink(
```
Comment (srueg) [inline commodore/dependency_mgmt.py:28]: We should probably still have some check if `class/defaults.yml` exists, otherwise we'll create broken symlinks.

**Review by srueg — CHANGES_REQUESTED** [review]

File: commodore/dependency_mgmt.py (hunk 1 above)
Comment (corvus-ch) [reply commodore/dependency_mgmt.py:28]: I have added a check in `relsymlink`. This probably was a general issue.

**Review by simu — APPROVED** [review]

File: commodore/helpers.py [hunk 2]
```diff
@@ -184,6 +184,10 @@ def relsymlink(src: P, dest_dir: P, dest_name: Optional[str] = None):
     # https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to
     link_src = os.path.relpath(src, start=dest_dir)
     link_dst = dest_dir / dest_name
+    if not P(link_src).exists():
```
Comment (srueg) [inline commodore/helpers.py:187]: Isn't `link_src` already a Path?
```suggestion
    if not link_src.exists():
```

File: commodore/helpers.py [hunk 3]
```diff
@@ -184,6 +184,10 @@ def relsymlink(src: P, dest_dir: P, dest_name: Optional[str] = None):
...
+        click.ClickException(
+            f"Can not lin {link_src} to {link_dst}. Source does not exist."
```
Comment (srueg) [inline commodore/helpers.py:189]:
```suggestion
            f"Can't link {link_src} to {link_dst}. Source does not exist."
```

**Review by srueg — CHANGES_REQUESTED** [review]

File: commodore/helpers.py (hunk 2 above)
Comment (corvus-ch) [reply commodore/helpers.py:187]: `relpath` returns type `Text`.

**Review by srueg — APPROVED** [review]

**Pull request merged**